assembl/views/api2/auth.py

```diff
@@ -368,31 +368,30 @@ def reset_password(request):
                     localizer.translate(_("This email does not exist")),
                     code=HTTPNotFound.code)
             else:
-                raise JSONError(
-                    localizer.translate(_(generic_error_message)),
-                    code=HTTPNotFound.code)
+                send_change_password_email(request, user, email, discussion=discussion)
                 logger.error("This email does not exist.")
         if account:
             email = account.email
     else:
         error = localizer.translate(_("Please give an identifier"))
         raise JSONError(error)
     if not email:
-        email = user.get_preferred_email()
+        if user:
+            email = user.get_preferred_email()
+        else:
+            email = None
         if not email:
             if not discussion.preferences['generic_errors']:
                 error = localizer.translate(_("This user has no email"))
             else:
-                error = localizer.translate(_(generic_error_message))
                 logger.error("This user has no email.")
-            raise JSONError(error, code=HTTPPreconditionFailed.code)
+                send_change_password_email(request, user, email, discussion=discussion)
     if not isinstance(user, User):
         if not discussion.preferences['generic_errors']:
             error = localizer.translate(_("This is not a user"))
         else:
-            error = localizer.translate(_(generic_error_message))
             logger.error("This is not a user.")
-        raise JSONError(error, code=HTTPPreconditionFailed.code)
+            send_change_password_email(request, user, email, discussion=discussion)
```

Comment:
use `return HTTPOk()` instead? and revert your change in `send_change_password_email`